### PR TITLE
Added Project Euler Problem 92: Fixes TheAlgorithms#2695

### PR DESCRIPTION
**Pull request by amy-graham-js**
### **Describe your change:**
This is the optimised solution for https://github.com/TheAlgorithms/Python/pull/3200 that @dhruvmanila requested

I have added a solution to Project Euler Problem 92: Square digit chains.
It prints the solution for the default limit 10,000,000.

I ran doctests locally, black and flake8 and they have all passed.

I would like this to contribute towards hacktoberfest please. Thanks

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by amy-graham-js** [comment]
@dhruvmanila please can you mark this as hacktoberfest?

**Review by 53jk1 — APPROVED** [review]

File: project_euler/problem_092/sol1.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    A number chain is created by continuously adding the square of the digits in a
+    number to form a new number until it has been seen before.
+    >>> solution(10)
+    7
+    >>> solution(350)
+    295
+    >>> solution(22222)
+    18803
+    >>> solution(10000000)
+    8581146
```
Comment (dhruvmanila) [inline project_euler/problem_092/sol1.py:28]: Please read point 5 from project_euler/README.md

File: project_euler/problem_092/sol1.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        chain = [i]
+        while True:
+            if i in squared:
+                if squared[i] == 89:
+                    squared.update(dict.fromkeys(chain, 89))
+                    eighty_nine_count += 1
+                    break
+                else:
+                    squared.update(dict.fromkeys(chain, 1))
+                    break
```
Comment (dhruvmanila) [inline project_euler/problem_092/sol1.py:43]: Combine the two `break` statements into one.

File: project_euler/problem_092/sol1.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    7
+    >>> solution(350)
+    295
+    >>> solution(22222)
+    18803
+    >>> solution(10000000)
+    8581146
+    """
+
+    squared = {1: 1, 89: 89}
```
Comment (dhruvmanila) [inline project_euler/problem_092/sol1.py:31]: If this is a cache then use `functools.lru_cache`

File: project_euler/problem_092/sol1.py (hunk 1 above)
Comment (amy-graham-js) [reply project_euler/problem_092/sol1.py:28]: Thanks, will implement these changes at the weekend

**Comment by amy-graham-js** [comment]
@dhruvmanila Thanks for your feedback, I have implemented the changes

File: project_euler/problem_092/sol1.py (hunk 3 above)
Comment (dhruvmanila) [reply project_euler/problem_092/sol1.py:32]: Ah, my bad. This was a mistake on my side. Please remove the `lru_cache` as it is not needed here. But try to optimize this solution as it is taking 50 sec to run: https://github.com/TheAlgorithms/Python/pull/3509/checks?check_run_id=1360369919#step:5:25

Comment (amy-graham-js) [reply project_euler/problem_092/sol1.py:29]: Sorry I have been very busy but I will try and optimise this weekend!

**Comment by amy-graham-js** [comment]
@dhruvmanila I have optimised the solution now and it is much faster. Thanks for your patience :)

**Pull request closed without merging**